pkg/pillar/cmd/zedagent/handlemetrics.go

```diff
@@ -1624,15 +1625,28 @@ func getDefaultRouters(ifname string) []string {
 func getAppIP(ctx *zedagentContext, aiStatus *types.AppInstanceStatus,
 	vifname string) (string, bool, string) {
 
+	sub := ctx.subNetworkInstanceStatus
 	log.Debugf("getAppIP(%s, %s)\n", aiStatus.Key(), vifname)
 	for _, ulStatus := range aiStatus.UnderlayNetworks {
 		if ulStatus.Vif != vifname {
 			continue
 		}
+		appIP := ulStatus.AllocatedIPAddr
+		ulconfig := ulStatus.UnderlayNetworkConfig
+		st, _ := sub.Get(ulconfig.Network.String())
+		if st != nil {
+			netstatus := cast.CastNetworkInstanceStatus(st)
+			if netstatus.Type == types.NetworkInstanceTypeSwitch {
+				log.Infof("getAppIP: type switch on %s", netstatus.BridgeName)
+				if _, ok := netstatus.IPAssignments[ulStatus.Mac]; ok {
+					appIP = netstatus.IPAssignments[ulStatus.Mac].String()
+					log.Infof("getAppIP: got sw net ip of app %s", appIP)
```

Comment:
Same here.. do we really need this to be an Info msg? 

These periodic messages clutter the the debug log output..

Reply:
will do.

Reply:
BTW, this upload of info is only called once, that's why I need to have an explicit trigger for upload again;-)